third_party/move/move-compiler-v2/src/pipeline/unused_assignment_checker.rs

```diff
@@ -17,41 +17,81 @@ use move_stackless_bytecode::{
-    /// Check if the assignment to `dst` is used at offset after the position given by `offset` and `after`.
+    /// Check if the assignment to `dst` is used after the position given by `offset`.
     fn check_unused_assignment(
         target: &FunctionTarget,
         id: AttrId,
         offset: CodeOffset,
         dst: TempIndex,
     ) {
         let loc = target.get_bytecode_loc(id);
-        // skip inlined code
+        // Skip inlined code.
         if loc.is_inlined() {
             return;
         }
-        let data = target.data;
-        // only check for user defined variables
-        if let Some(dst_name) = data.local_names.get(&dst) {
-            let live_var_info = target
-                .get_annotations()
-                .get::<LiveVarAnnotation>()
-                .expect("live variable annotation")
-                .get_info_at(offset);
+        let live_var_annotations = target
+            .get_annotations()
+            .get::<LiveVarAnnotation>()
+            .expect("live variable annotation");
+        // Only check for user defined variables.
+        if let Some(dst_name) = target.get_local_name_opt(dst) {
+            let live_var_info = live_var_annotations.get_info_at(offset);
             let live_after = &live_var_info.after;
-            let dst_name = dst_name.display(target.func_env.symbol_pool()).to_string();
             if !dst_name.starts_with('_') && live_after.get(&dst).is_none() {
                 target.global_env().diag(
                     Severity::Warning,
                     &loc,
                     &format!(
-                        "Unused assignment/binding to `{}`. Consider removing the assignment/binding, \
-                        or prefixing with an underscore (e.g., `_{}`), or binding to `_`",
+                        "This assignment/binding to the left-hand-side variable `{}` is unused. \
+                        Consider removing this assignment/binding, \
+                        or prefixing the left-hand-side variable with an underscore (e.g., `_{}`), or renaming to `_`",
                         dst_name, dst_name
                     ),
                 );
             }
         }
     }
+
+    /// Check if `target` function's parameters are unused.
+    fn check_params(target: &FunctionTarget) {
+        let live_after_first_instr = &target
+            .get_annotations()
+            .get::<LiveVarAnnotation>()
+            .expect("live variable annotation should be present")
+            .get_info_at(0 as CodeOffset)
+            .after;
+        let sources_of_first_instr = target.get_bytecode()[0].sources();
+        let dests_of_first_instr = target.get_bytecode()[0].dests();
+        let params = target.func_env.get_parameters();
+        for (param_index, param) in params.iter().enumerate() {
+            if let Some(param_name) = target.get_local_name_opt(param_index) {
+                if !param_name.starts_with('_') && param_name != "self" {
```

Comment:
nit: maybe use RECEIVER_PARAM_NAME instead of "self" here?

Reply:
done